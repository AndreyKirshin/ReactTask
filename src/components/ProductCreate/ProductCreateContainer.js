import React, { Fragment, useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Header from '../Header/Header';
import { fetchCategories } from '../../actions/categories';
import { fetchProducts, createProduct } from '../../actions/products';
import ProductForm from '../ProductForm/ProductForm';
import 'react-notifications/lib/notifications.css';
import {NotificationContainer, NotificationManager} from 'react-notifications';

export const ProductCreateContainer = ({ categories, getCaregories, getProducts, createProduct, notifications }) => {
  useEffect(() => {
    getCaregories();
    getProducts();
  }, [getCaregories, getProducts]);

  useEffect(() => {
    if (notifications) {
      NotificationManager.success('Product has been added', 'Success!');
    }
  }, [notifications]);
  
  return (
    <Fragment>
      <Header name="Create Product"/>
      <div>we will create here</div>
      <ProductForm categories={categories} sendProduct={createProduct}/>
      <NotificationContainer/>
    </Fragment>
  )
}

ProductCreateContainer.propTypes = {
  categories: PropTypes.array.isRequired,
  getCaregories: PropTypes.func.isRequired,
  getProducts: PropTypes.func.isRequired,
  createProduct: PropTypes.func.isRequired,
  notifications: PropTypes.bool.isRequired
};

const mapStateToProps = state => {
  const { categories, notifications } = state;
  return {
    categories,
    notifications
  }
};

const mapDispatchToProps = dispatch => ({
  getCaregories: () => dispatch(fetchCategories()),
  getProducts: () => dispatch(fetchProducts()),
  createProduct: product => dispatch(createProduct(product))
});
  

export default connect(mapStateToProps, mapDispatchToProps)(ProductCreateContainer);