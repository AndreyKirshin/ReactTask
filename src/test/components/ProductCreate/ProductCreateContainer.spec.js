import React from 'react';
import { ProductCreateContainer } from '../../../components/ProductCreate/ProductCreateContainer';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


describe('<ProductCreateContainer/>', () => {
  const props = {
    categories: [],
    getCaregories: jest.fn(),
    getProducts: jest.fn(),
    createProduct: jest.fn(),
    notifications: false
  }
  it('should match snapchot', () => {
    const wrapper = shallow(<ProductCreateContainer {...props}/>);

    expect(wrapper).toMatchSnapshot()
  })
})