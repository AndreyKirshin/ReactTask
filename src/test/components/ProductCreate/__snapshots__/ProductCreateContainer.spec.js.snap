// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCreateContainer/> should match snapchot 1`] = `ShallowWrapper {}`;
